addon/components/ui-text-input/component.ts

```diff
@@ -21,10 +23,27 @@ export default class UiTextInput extends Component {
   disabled?: boolean = false;
   max?: number | null = null;
   min?: number | null = null;
+  width?: string; //Accepts any standard CSS width measurement
 
   @localClassName()
   hasError?: boolean = false;
 
+  @attribute()
+  @computed("width")
+  get style(): SafeString {
+    if (this.width) {
+      return htmlSafe(`width:${this.width}`);
+    } else {
+      return htmlSafe("");
+    }
+  }
+
+  @localClassName()
+  @computed("width")
+  get hasWidth(): boolean {
+    return !!this.width;
+  }
+
```

Comment:
Maybe better to do this in the template? E.g.

`{{input class=(if (not-empty this.width "has-width")) ...}}`

Reply:
`not-empty` isn't a helper, so I have done `(if (not (is-empty this.width)) "has-width")` instead